casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -210,25 +210,20 @@ class RuntimeManagerImpl[F[_]: Concurrent] private[rholang] (
       ConstructDeploy.sourceDeployNow(deployPaymentSource(amount)).withDeployer(user)
     ) >>
       getResult(runtime)() >>= {
-      case Seq(RhoType.Boolean(true)) =>
+      case Seq(RhoType.Tuple2(RhoType.Boolean(true), Par.defaultInstance)) =>
         runtime.space.createCheckpoint()
-      case Seq(RhoType.String(error)) =>
+      case Seq(RhoType.Tuple2(RhoType.Boolean(false), RhoType.String(error))) =>
         BugFoundError(s"Deploy payment failed unexpectedly: $error").raiseError[F, Checkpoint]
       case other =>
+        val contentAsStr = other.map(RholangPrinter().buildString(_)).mkString(",")
         BugFoundError(
-          s"Deploy payment returned unexpected result: ${other.map(RholangPrinter().buildString(_))}"
+          s"Deploy payment returned unexpected result: [$contentAsStr ]"
         ).raiseError[F, Checkpoint]
     }
 
   private def deployPaymentSource(amount: Long, name: String = "__SCALA__"): String =
-    s"""
-       | new rl(`rho:registry:lookup`), poSCh in {
-       |   rl!(`rho:id:cnec3pa8prp4out3yc8facon6grm3xbsotpd4ckjfx8ghuw77xadzt`, *poSCh) |
-       |   for(@(_, PoS) <- poSCh) {
-       |     @PoS!("pay", $amount, "$name")
-       |   }
-       | }
-       """.stripMargin
+    s"""@"$name"!((true, Nil))"""
```

Comment:
Why was this eliminated/changed? Why are you planning to pull `PoS` from `SystemInstances`? It should be taken from the registry like the existing implementation suggests. Please revert this change.

Reply:
No. The existing code does not rely on `SystemRegistry`. The changes you made should leave everything in tact, including the RuntimeManager. If your changes make something break in RuntimeManager, then there's something wrong with your changes. Please revert this and all other changes to RuntimeManager.